crates/iota-core/src/execution_cache/unit_tests/writeback_cache_tests.rs

```diff
@@ -1328,6 +1328,62 @@ async fn latest_object_cache_race_test() {
     invalidator.join().unwrap();
 }
 
+#[tokio::test]
+async fn test_transaction_cache_race() {
```

Comment:
Does this test trigger "entry is newer than value" assert? Did this test fail before this commit? I'd leave an explainer comment.

Reply:
The fix is moving store insertion before cache insertions. "entry is newer than value" doesn't relate to the bug 😄